app/policies/idv/step_info.rb

```diff
@@ -29,5 +30,11 @@ def preconditions_validation
         errors.add(:preconditions, type: :invalid_argument, message: 'preconditions must be a Proc')
       end
     end
+
+    def undo_step_validation
```

Comment:
Does this need to be wired in using `validate`?

Reply:
This caused an error in specs when we hadn't added undo_step to :root, so it's working as intended so far. What would you suggest?

Reply:
If it's not wired into validation, then it won't get executed. I think if you didn't have `undo_step:` on :root you'd get an error but just because an argument was missing, not because it was the wrong type.

Reply:
Fixed, thanks for pairing!